src/screenshots/capturer.js

```diff
@@ -101,12 +102,15 @@ export default class Capturer {
         return joinPath(imageDir, 'thumbnails', imageName);
     }
 
-    async _takeScreenshot (filePath, pageWidth, pageHeight) {
+    async _takeScreenshot (filePath, pageWidth, pageHeight, fullPage) {
         await makeDir(dirname(filePath));
-        await this.provider.takeScreenshot(this.browserId, filePath, pageWidth, pageHeight);
+
+        fullPage = fullPage !== void 0 ? fullPage : this.fullPage;
```

Comment:
Simplify as follows:
```js
async _takeScreenshot (filePath, pageWidth, pageHeight, fullPage = false) {
    ....
    fullPage = fullPage || this.fullPage;
}
```

Reply:
here we need check that `undefined` !== false
the `fullPage` option has 3 states: true, false, undefined.
If it is undefined, then we should get the value from config
If it is false, then we should ignore the value from config even if it's true there

Reply:
I guess you can just use 
```
async _takeScreenshot (filePath, pageWidth, pageHeight, fullPage = this.fullPage) 
```

Reply:
> here we need check that `undefined` !== false
> the `fullPage` option has 3 states: true, false, undefined.
> If it is undefined, then we should get the value from config
> If it is false, then we should ignore the value from config even if it's true there

It occurred that we can use command with bool property but `undefined` value as default. It works beautifully